generate-errors/src/lib.rs

```diff
@@ -1,114 +1,213 @@
+        for line in regex_content.lines() {
+            // Ensure we only operate on
+            // Rust code blocks.
+            if !line.starts_with("```rust") {
+                content.push_str(line);
+                content.push('\n');
                 continue;
             }
 
-            let error_code = read_to_string(path.clone())?;
-
-            let code = file_name.trim_end_matches(".md").to_owned();
-            section.content.push(ErrorCode {
-                content: error_code
-                    .trim_start_matches(&format!("# {}", code.clone()))
-                    .replace("```rust,*", "```rust")
-                    .lines()
-                    .map(|line| {
-                        // throw away `should_panic` and `no_run` to fix code highlighting
-                        if line.starts_with("```rust,") {
-                            "```rust"
-                        } else {
-                            line
-                        }
-                    })
-                    .collect::<Vec<&str>>()
-                    .join("\n"),
-                code,
-            });
+            let annotations = line
+                .strip_prefix("```")
+                .ok_or(anyhow!(
+                    "Failed to find start of code block to strip from string"
+                ))?
+                .split(',');
+
+            let mut line: String = String::from("```");
+
+            // Add all annotations other than rust
+            // to the content first to avoid the issue.
+            for annotation in annotations {
+                if annotation == "rust" {
+                    continue;
+                }
+
+                line.push_str(annotation);
+                line.push(',');
+            }
+            line.push_str("rust");
+
+            content.push_str(&line);
+            content.push('\n');
         }
+
+        results_map.insert(file_name, content.to_string());
     }
 
-    section.content.sort_by(|a, b| a.code.cmp(&b.code));
+    Ok(results_map)
+}
+
+/// Writes a valid docs section to contain
+/// the error pages in.
+///
+/// The output path passed should be the folder
+/// you want the Zola pages to be written / output.
+pub fn write_section(output_path: &Path) -> anyhow::Result<()> {
+    let errors_folder_path = output_path.join("errors");
+    // Make sure the output folder exists
+    fs::create_dir_all(&errors_folder_path)?;
+
+    let section_content = r#"+++
+title = "Errors"
+template = "docs-section.html"
+page_template = "docs-page.html"
+redirect_to = "/learn/errors/introduction"
++++
+"#;
+
+    let introduction_content = r#"+++
+title = "Introduction"
+[extra]
+weight = 0
++++
+
+These pages document Bevy's error codes for the _current release_.
+
+In case you are looking for the latest error codes from Bevy's main branch, you can find them in the [Bevy engine repository](<https://github.com/bevyengine/bevy/tree/main/errors>). 
+"#;
```

Comment:
These can probably be converted into `const`s, but I won't block on it. This function only gets called once and Rust likely optimizes it anyways.

Reply:
Switching to const breaks stuff because we use non-constant values and error handling in the formatting. Not worth it imo.